### DISCUSSION TITLE
Change code background color in PDF document?

### DISCUSSION THREAD

**Discussion by maxdrohde** [quarto-dev/quarto-cli#607] [Q&A] [answered]
After downloading the new version of Quarto, it seems that the default background color for code in PDF documents has changed from blue to white. See below for an example.

Does anyone know how to manually set the default background color for code? I tried looking on this page but couldn't find any options: https://quarto.org/docs/reference/formats/pdf.html

### New

<img width="800" alt="Screen Shot 2022-04-10 at 19 32 55" src="https://user-images.githubusercontent.com/25734232/162647135-3218f4ef-1004-4d33-bc07-0718f9f6a3ca.png">

### Old

<img width="800" alt="Screen Shot 2022-04-10 at 19 32 34" src="https://user-images.githubusercontent.com/25734232/162647139-4cff7329-8014-49f5-ae2e-9cfedb34e067.png">

**Comment by dragonstyle** [comment, marked as answer]
Great question (it was a little challenging to find in our docs, maybe something we can improve). More is here:

https://quarto.org/docs/authoring/article-layout.html#code-blocks

if you set `code-block-border-left: false`, it will go back to the background color treatment...

For example:

```
format:
  pdf:
    code-block-border-left: false
```

**Reply by maxdrohde** [reply]
Got it, good to know. I'm able to restore the previous background color using `code-block-border-left: false`

However, I can't change the background color with `code-block-bg:` in PDF documents like I can in HTML documents -- is this intended?

**Reply by maxdrohde** [reply]
Also, I think my confusion with the documentation was because I was expecting all possible code formatting options to be under the code heading on the PDF reference page here: https://quarto.org/docs/reference/formats/pdf.html

**Reply by dragonstyle** [reply]
Today the background color doesn't respect `code-block-bg`, but I agree that it would make sense for it to do so. I can convert this thread into an issue and try to get that implemented soon (or if you're feeling ambitious, happy to accept a PR :) ).

Agree about the documentation - we'll refine that too (the fact that appeared in article layout makes it seem like it only applies to documents with advanced layout, as well, which isn't true)...

**Reply by dragonstyle** [reply]
https://github.com/quarto-dev/quarto-cli/issues/608

and

https://github.com/quarto-dev/quarto-web/issues/118

Thank you for asking - we'll improve this!

**Reply by maxdrohde** [reply]
Thanks, I appreciate the work you all do!

**Reply by dkStevensNZed** [reply]
I'm a little late to the game but after reading this I tried the solution but the code blocks still have no shading (in any color :) in my pdf document. My front matter is below, Any thoughts?
---
title: "Foobar"
author:
  - name: Me
    degrees: 
      - B.S.
      - PhD
      - PE
    affiliation: 
      - Where I work
format: 
  pdf:
    toc: true
    toc-depth: 3
    lof: true
    lot: true
    listings: true
    code-block-border-left: false
    header-includes:
    - \usepackage{float}
    - \floatplacement{table}{H}
    - \usepackage[justification=raggedright,labelfont=bf,singlelinecheck=false,labelformat=empty]{caption}
geometry: 
  - left=  25mm
  - right= 25mm
  - top=   25mm
  - bottom=10mm
mainfont: Lato
mathfont: Roboto
bibliography: refs.bib
csl: american-society-of-civil-engineers.csl
---

**Reply by mcanouil** [reply]
@dkStevensNZed Could you share a fully (and small) reproducible example? yours is not in addition to not be formatted corrected inside a markdown code blocks. Thank you.

Note that this discussion was opened/closed before Quarto was officially release.
Since then there were a lot of internal changes including a a major upgrade of Pandoc the underlying tool used by Quarto.
So, the solution might need some tweaks.

**Reply by dkStevensNZed** [reply]
Sorry about that. Here's a simple example. When I render to html, the code box is shaded. When I render to pdf, there's no shading. For unknown reasons, the indentation in the YAML is lost when the message is posted. 

**_New info_**: If I remove the format/pdf lines  
  
    toc: true
    toc-depth: 3
    lof: true
    lot: true
    listings: true

the shading is working in pdf. With these lines, there is no shading.

---
title: "Foobar"
author: Me
format: 
  html: default
  pdf:
    toc: true
    toc-depth: 3
    lof: true
    lot: true
    listings: true
    code-block-border-left: false
    header-includes:
    - \usepackage{float}
    - \floatplacement{table}{H}
    - \usepackage[justification=raggedright,labelfont=bf,singlelinecheck=false,labelformat=empty]{caption}
geometry: 
  - left=  25mm
  - right= 25mm
  - top=   25mm
  - bottom=10mm
mainfont: Lato
mathfont: Roboto
---

Here's a code box

```{r}
d <- 1
print(d)
```

**Reply by dkStevensNZed** [reply]
Update. It appears that the 

`listings: true
`
is causing the problem. Without that line, the shading appears, even with the toc/lof/lot included

**Reply by mcanouil** [reply]
@dkStevensNZed Could you edit and format properly your posts? They are currently unusable. If your code include code blocks, you need to use more backticks on the enclosing code blocks.

**Reply by dkStevensNZed** [reply]
The attached file contains TestCodeBox.qmd (quarto document) that compiles but does not shade the code boxes.

[TestCodeBox.zip](https://github.com/quarto-dev/quarto-cli/files/14114953/TestCodeBox.zip)
